packages/synapse-react-client/src/lib/containers/auth/TwoFactorSecretDialog.tsx

```diff
@@ -0,0 +1,64 @@
+import {
+  Box,
+  Button,
+  Dialog,
+  DialogActions,
+  DialogContent,
+  DialogProps,
+  DialogTitle,
+  IconButton,
+  Stack,
+  Typography,
+} from '@mui/material'
+import IconSvg from '../IconSvg'
+import React from 'react'
+
+type TwoFactorSecretDialogProps = {
+  secret: string
+} & Pick<DialogProps, 'open' | 'onClose'>
+
+export default function TwoFactorSecretDialog(
+  props: TwoFactorSecretDialogProps,
+) {
+  const { secret, open, onClose } = props
+  return (
+    <Dialog open={open} onClose={onClose}>
+      <DialogTitle>
+        <Stack direction="row" alignItems={'center'} gap={'5px'}>
+          Use this code to set up 2FA <Box sx={{ flexGrow: 1 }} />
+          <IconButton aria-label={'Close'} onClick={onClose}>
+            <IconSvg icon={'close'} wrap={false} sx={{ color: 'grey.700' }} />
+          </IconButton>
+        </Stack>
+      </DialogTitle>
+      <DialogContent>
+        <Typography variant={'body1'}>
+          Your authenticator app will ask for a code to activate two-factor
+          authentication. Use the code below:
+        </Typography>
+        <Box
+          sx={theme => ({
+            my: 2,
+            p: 2,
+            width: '100%',
+            textAlign: 'center',
+            background: theme.palette.grey[200],
+            fontFamily: 'Menlo, monospace',
```

Comment:
Menlo was used in the Figma, not sure if we should add it to the list of fonts to download.